io/EptReader.cpp

```diff
@@ -767,5 +775,88 @@ Dimension::Type EptReader::getCoercedTypeTest(const NL::json& j)
     return getCoercedType(j);
 }
 
+//============================================================================================
+// Pravin: APIs from here are written to make EPT Reader streamable.
+
+void EptReader::loadNextOverlap()
+{
+    Key key(m_keys.at(0));
+    m_keys.erase(m_keys.begin());
+    log()->get(LogLevel::Debug)
+        << "Streaming Data " << m_nodeId << "/" << m_overlaps.size() << ": "
+        << key.toString() << std::endl;
+
+    uint64_t startId(0);
+	m_pointTable.reset(new PointTable());
+	if (m_info->dataType() == EptInfo::DataType::Laszip)
+    {
+		m_pointView.reset(new PointView(*m_pointTable));
+		startId=readLaszip(*m_pointView, key, m_nodeId, m_pointTable);
+    }
+    else
+    {
+        std::unique_lock<std::mutex> lock(m_mutex);
+        prepare(*m_pointTable);
+        lock.unlock();
+        m_pointView.reset(new PointView(*m_pointTable));
+        startId = readBinary(*m_pointView, key, m_nodeId);
+	}
+    m_currentIndex = 0;
+
+    // Read addon information after the native data, we'll possibly
+    // overwrite attributes.
+    for (const auto& addon : m_addons)
+    {
+        readAddon(*m_pointView, key, *addon, startId);
+    }
+    m_nodeId++;
+}
+
+void EptReader::saveOverlapKeys()
+{
+    for (const auto& entry : m_overlaps)
+    {
+        const Key& key(entry.first);
+        m_keys.push_back(key);
+    }
+}
+
+void EptReader::fillPoint(PointRef& point)
+{
+    DimTypeList dims = m_pointView->dimTypes();
```

Comment:
could you not simply set the PointRef to point to the current in the view as oppposed to cpoying the data?

Reply:
`=` operator is deleted from `PointRef`, so we cannot directly set it.
We've to copy.